tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,326 @@
+	}
+	const backupName = "backup1"
+	backends := []string{"Minio", "S3"}
+	for _, backend := range backends {
+		t.Run(fmt.Sprintf("Medusa on %s", backend), func(t *testing.T) {
+			namespace := initializeCluster(t)
+			createMedusaSecretAndInstallDeps(t, namespace, backend)
+			deployClusterForMedusa(t, namespace, backend)
+			medusaSuccess := t.Run("Test backup and restore", func(t *testing.T) {
+				testMedusa(t, namespace, backend, backupName)
+			})
+			if medusaSuccess {
+				cleanupCluster(t, namespace)
+			}
+		})
+	}
+}
+
+func testMedusa(t *testing.T, namespace, backend, backupName string) {
+	log.Println(Step("Testing Medusa..."))
+	log.Println("Creating test keyspace and table")
+	CreateCassandraTable(t, namespace, medusaTestTable, medusaTestKeyspace)
+
+	loadRowsAndCheckCount(t, namespace, 10, 10)
+
+	log.Println(Info("Backing up Cassandra"))
+	PerformBackup(t, namespace, backupName)
+
+	loadRowsAndCheckCount(t, namespace, 10, 20)
+
+	log.Println(Info("Restoring backup and checking row count"))
+	RestoreBackup(t, namespace, backupName)
+	CheckRowCountInTable(t, 10, namespace, medusaTestTable, medusaTestKeyspace)
+}
+
+func deployClusterForMedusa(t *testing.T, namespace, backend string) {
+	log.Println(Info(fmt.Sprintf("Deploying K8ssandra with Medusa using %s", backend)))
+	valuesFile := fmt.Sprintf("one_node_cluster_with_medusa_%s.yaml", strings.ToLower(backend))
+	DeployClusterWithValues(t, namespace, strings.ToLower(backend), valuesFile)
+	CheckClusterExpectedResources(t, namespace)
+}
+
+func loadRowsAndCheckCount(t *testing.T, namespace string, rowsToLoad, rowsExpected int) {
+	log.Println(Info(fmt.Sprintf("Loading %d rows and checking we have %d after", rowsToLoad, rowsExpected)))
+	LoadRowsInTable(t, rowsToLoad, namespace, medusaTestTable, medusaTestKeyspace)
+	CheckRowCountInTable(t, rowsExpected, namespace, medusaTestTable, medusaTestKeyspace)
+}
+
+func createMedusaSecretAndInstallDeps(t *testing.T, namespace, backend string) {
+	log.Println(Info("Creating medusa secret to access the backend"))
+	if backend == "Minio" {
+		DeployMinioAndCreateBucket(t, "k8ssandra-medusa")
+		CreateMedusaSecretWithFile(t, namespace, "secret/medusa_minio_secret.yaml")
+	} else {
+		// S3
+		CreateMedusaSecretWithFile(t, namespace, "~/medusa_secret.yaml")
+	}
+}
+
+// Monitoring scenario:
+// - Install Traefik
+// - Create a namespace
+// - Register a cluster with three Cassandra nodes and one Stargate node
+// - Check that Prometheus is reachable through its REST API
+// - Check the number of active Prometheus targets
+// - Check that Grafana is reachable through http
+// - Terminate the namespace and delete the cluster
+func TestMonitoringDeploymentScenario(t *testing.T) {
+	// only runs when the -short flag is used
+	if !testing.Short() {
+		t.Skip("Skipped in non-short mode")
+	}
+	namespace := initializeCluster(t)
+	deployClusterForMonitoring(t, namespace)
+	prometheusSuccess := t.Run("Test Prometheus", func(t *testing.T) {
```

Comment:
What's the point of t.Run in this case? Why not just testPrometheus(t, namespace) ?

Reply:
For this specific test scenario, it will allow to test Grafana even if the Prometheus test fails.
I can remove the subtest for the Stargate and Reaper tests though since there's a single component that is being tested there.

Reply:
Sounds good to you @burmanm ?

Reply:
Since the cleanup phase is now configurable and depends on both config and test success, I've had to wrap all tests within subtests.
I bet there are other ways to achieve this, but such changes would have to be done in a follow up PR.